pallets/gear/src/lib.rs

```diff
@@ -602,6 +604,15 @@ pub mod pallet {
                 Error::<T>::ProgramAlreadyExists
             );
 
+            // Create an account for the program
+            CurrencyOf::<T>::transfer(
+                &who,
+                &program_id.cast(),
+                CurrencyOf::<T>::minimum_balance(),
+                ExistenceRequirement::KeepAlive,
+            )?;
+            System::<T>::inc_consumers(&program_id.cast())?;
```

Comment:
why not locking balance?

Reply:
It's essentially as good as creating a lock to ensure that an account will always exist. But more efficient as there is no need to iterate through existing locks when adding a new one.

Reply:
**Update:** as discussed, replaced the direct consumers increment for a program account with a lock for the existential deposit. In the view of that all the arguments against this further down in this PR can be ignored as obsolete.